test/testhelpers/assert.go

```diff
@@ -71,12 +71,34 @@ func AssertReportResultsEqual(t *testing.T, expected, actual []map[string]interf
 	}
 }
 
-func AssertErrorContainsErrorMsgs(t *testing.T, err error, errMsgArr []string) {
+func AssertErrorContainsErrorMsgs(t *testing.T, err error, errMsgArr []string) bool {
```

Comment:
I'd kinda like to see less use of this type of assertion, not more. We're asserting on specific strings making the tests more brittle, if someone comes along and changes an error message, we've got a failing test and not really for a good reason. One thing to do might be to adopt error types. That way we can assert on the error type and not the specific message. Maybe not for this PR, but just to keep in mind.

Reply:
That makes sense to me, and I have updated the error messages in the `deploy` package before, which in turn would break a test's expected error messages. As a stop-gap, we decided to use a series of regexes instead of strings, which helps in the aforementioned case above, or in the case where the apiserver is returning a different error message than what we expected (namely due to differing kube versions.)

Reply:
I can add a `TODO: ...` comment that links back to your concerns so we can rethink how we handle error messages further down the line.

Reply:
yeah mention error types in the TODO -- I think that is the main logic to having error types in a testing regime, so you can assert on them and not error strings.